Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/create/TraceLocationCreateViewModel.kt

```diff
@@ -0,0 +1,125 @@
+package de.rki.coronawarnapp.ui.eventregistration.organizer.create
+
+import android.content.res.Resources
+import androidx.annotation.StringRes
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.contactdiary.util.CWADateTimeFormatPatternFactory.shortDatePattern
+import de.rki.coronawarnapp.ui.durationpicker.toReadableDuration
+import de.rki.coronawarnapp.ui.eventregistration.organizer.category.adapter.category.TraceLocationCategory
+import de.rki.coronawarnapp.ui.eventregistration.organizer.category.adapter.category.TraceLocationUIType
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import org.joda.time.Duration
+import org.joda.time.LocalDateTime
+import java.util.Locale
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class TraceLocationCreateViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    @Assisted private val category: TraceLocationCategory
+) :
+    CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    private val mutableUiState = MutableLiveData<UIState>()
+    val uiState: LiveData<UIState>
+        get() = mutableUiState
+
+    var description: String? by UpdateDelegate()
+    var address: String? by UpdateDelegate()
+    var checkInLength: Duration? by UpdateDelegate()
+    var start: LocalDateTime? by UpdateDelegate()
+    var end: LocalDateTime? by UpdateDelegate()
```

Comment:
The state of `checkInLength`, `start`, and `end` is lost on process death..

Reply:
What do you mean? `ViewModel` is scoped to `TraceLocationCreateFragment` so the data are kept in memory until the fragment is not destroyed. We don't need to preserve unfinished event data.

Reply:
Fixed for case when activity is killed and recreated